// @wix/secrets/build/es/src/velo-secrets-vault-v1-secret.meta.js
import * as ambassadorWixVeloSecretsVaultV1Secret from './velo-secrets-vault-v1-secret.http';
export function getSecretValue() {
    const payload = { name: ':name' };
    const getRequestOptions = ambassadorWixVeloSecretsVaultV1Secret.getSecretValue(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/api/v1/secrets/name/{name}',
        pathParams: { name: 'name' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listSecretInfo() {
    const payload = {};
    const getRequestOptions = ambassadorWixVeloSecretsVaultV1Secret.listSecretInfo(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/api/v1/secrets',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createSecret() {
    const payload = {};
    const getRequestOptions = ambassadorWixVeloSecretsVaultV1Secret.createSecret(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/api/v1/secrets',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteSecret() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixVeloSecretsVaultV1Secret.deleteSecret(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/api/v1/secrets/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateSecret() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixVeloSecretsVaultV1Secret.partiallyUpdateSecret(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/api/v1/secrets/{id}',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
